lightstep-developer-satellite.sh

```diff
@@ -1,69 +1,129 @@
   exit 1
 fi
 
-ID=$(docker ps | grep lightstep/collector | head -n 1 | cut -d ' ' -f 1 )
+ID=$(docker ps | grep ${IMAGE} | head -n 1 | cut -d ' ' -f 1 )
 if [ -n "$ID" ]; then
   echo "There is already a lightstep collector running.  You can stop it by"
   echo "running stop-lightstep-collector.sh"
   exit 1
 fi
 
-if [ -z "$LIGHTSTEP_API_KEY" ]; then 
-  echo "LIGHTSTEP_API_KEY is not set.  You can retrieve an API key from"
-  echo "http://app.lightstep.com/<your project>/account"
-  echo "Please enter an API key:"
-  read -r LIGHTSTEP_API_KEY
-  echo "Thank you. In the future you may set LIGHTSTEP_API_KEY to skip this step."
-fi
-
 if [ -z "$LIGHTSTEP_USER" ]; then 
   echo "LIGHTSTEP_USER is not set.  This is the e-mail address you use for sign-in."
   echo "Please enter your user name:"
   read -r LIGHTSTEP_USER
   echo "Thank you. In the future you may set LIGHTSTEP_USER to skip this step."
 fi
 
-if [ -z "$LIGHTSTEP_PROJECT_ID" ]; then 
-  echo "LIGHTSTEP_PROJECT_ID is not set.  This must be set on the command-line."
-  exit 1
+if [ -z "$LIGHTSTEP_PROJECT" ]; then 
+  echo "LIGHTSTEP_PROJECT is not set.  This is the e-mail address you use for sign-in."
+  echo "Please enter your user name:"
+  read -r LIGHTSTEP_PROJECT
+  echo "Thank you. In the future you may set LIGHTSTEP_PROJECT to skip this step."
 fi
 
-## Set env vars to be passed to docker if not set in current environment.
-COLLECTOR_API_KEY="$LIGHTSTEP_API_KEY"
+if [ -z "$LIGHTSTEP_API_KEY" ]; then 
+  echo "LIGHTSTEP_API_KEY is not set.  You can retrieve an API key from"
+  echo "http://app.lightstep.com/${LIGHTSTEP_PROJECT}/developer-mode"
+  echo "Please enter an API key:"
+  read -r LIGHTSTEP_API_KEY
+  echo "Thank you. In the future you may set LIGHTSTEP_API_KEY to skip this step."
+fi
+
+# For certain configuration parameters, we want to set defaults if they haven't been set
+# in the environment.  Only the COLLECTOR_API_KEY is set unconditionally here.  All the
+# other COLLECTOR_ environment variables will override the values derived here, including
+# those derived from the LIGHTSTEP_* argument variables.
+
+COLLECTOR_API_KEY="${LIGHTSTEP_API_KEY}"
 # Developer-mode specifics
-: "${COLLECTOR_POOL:=${LIGHTSTEP_USER}_developer_pool}"
+: "${COLLECTOR_POOL:=${LIGHTSTEP_USER}_${LIGHTSTEP_PROJECT}_developer}"
+: "${COLLECTOR_PROJECT_NAME:=${LIGHTSTEP_PROJECT}}"
 : "${COLLECTOR_INGESTION_TAGS:=lightstep.developer:${LIGHTSTEP_USER}}"
 : "${COLLECTOR_DISABLE_ACCESS_TOKEN_CHECKING:=true}"
-: "${COLLECTOR_PROJECT_ID:=${LIGHTSTEP_PROJECT_ID}}"
 # Set default ports
-: "${COLLECTOR_BABYSITTER_PORT:=8000}"
-: "${COLLECTOR_ADMIN_PLAIN_PORT:=8080}"
-: "${COLLECTOR_HTTP_PLAIN_PORT:=8181}"
-: "${COLLECTOR_GRPC_PLAIN_PORT:=8282}"
-: "${COLLECTOR_PLAIN_PORT:=8383}"
+: "${COLLECTOR_GRPC_PLAIN_PORT:=8360}"
+: "${COLLECTOR_ADMIN_PLAIN_PORT:=8361}"
+# Disable unused ports
+: "${COLLECTOR_HTTP_PLAIN_PORT:=0}"
```

Comment:
If these are definitely intended to be disabled, I think they should not be exposed by docker below.

Reply:
Ah, that's fine then.  I now see that map_port handles zero ports to handle these, which is what I was confused about.